### ISSUE TITLE
kubectl-gs command to create a workload cluster key pair

### ISSUE THREAD

**Issue by marians** [giantswarm/roadmap#460]
## User story

- _As a customer admin, I want to create a key pair and kubectl config for accessing a workload cluster_

## Background

The preferred method for workload cluster access is OIDC authentication. However, in some cases -- for example for bootstrapping a cluster -- OIDC authentication is not available, and a fall-back method is needed.

As of now, customer admins use `gsctl create kubeconfig` and `gsctl create keypair` for creating and downloading key pairs, which uses the Rest API. As the Rest API and gsctl are deprecated, we need a method to create a workload cluster key pair via the Management API.

The internals of the method are explained [in our documentation](https://docs.giantswarm.io/ui-api/management-api/wc-key-pairs/).

A `kubectl gs` command should make this process more convenient and robust, and in addition allow to

- either store the key pair and workload cluster CA certificate as files
- or add the required `users`, `clusters`, and `contexts` entry to the user's kubectl config file (usually `~/.kube/config`)
- or create a separate kubectl config file with the key pair added as inline configuration values.

**Comment by marians** [comment]
For reference, there was a discussion regarding the spec in this Slack thread: https://gigantic.slack.com/archives/C13MG9F0W/p1631780643048200

**Comment by marians** [comment]
> It would be nice to be able to create a self-contained kubeconfig containing all the necessary data, including the CA cert as `certificate-authority-data`.

@agileman-droid To clarify, the resulting self-contained config file would only contain the access details for one workload cluster, correct? No management cluster OIDC details. So you can hand that to another person or use it in automations?

**Comment by snizhana-dynnyk** [comment]
parked for now, blocked by https://github.com/giantswarm/giantswarm/issues/19240
